google-datacatalog-qlik-connector/src/google/datacatalog_connectors/qlik/prepare/datacatalog_entry_factory.py

```diff
@@ -72,6 +72,53 @@ def make_entry_for_app(self, app_metadata):
 
         return generated_id, entry
 
+    def make_entry_for_custom_property_definition(
+            self, custom_property_def_metadata):
+
+        entry = datacatalog.Entry()
+
+        generated_id = self.__format_id(
+            constants.ENTRY_ID_PART_CUSTOM_PROPERTY_DEFINITION,
+            custom_property_def_metadata.get('id'))
+        entry.name = datacatalog.DataCatalogClient.entry_path(
```

Comment:
Is this ID safe?

By safe I mean let's say you have different custom properties at different levels of the APP, or across multiple APPs.

Won't the ID's overlap?

For the RDBMS connectors we need to prefix the table name with the schema name in the ID for that reason. Should something similar be done here?

Reply:
The Custom Property Definitions are managed at the Qlik site level, and can be applied to multiple objects hosted in that site, including Apps and Streams, just to let it clear. The ids are strings such as `a09ef343-29ea-4ff4-abd6-fe10cbc272ed`. I don't know how they are generated, but I don't believe they would overlap.